src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,82 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private Node[] elements;
+    private int size;
+
+    public StorageImpl() {
+        elements = new Node[10];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < size; i++) {
+            if (elements[i].getKey() == key
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                elements[i].setValue(value);
+                return;
+            }
+        }
+        Node element = new Node(key, value);
+        elements[size] = element;
+        size++;
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < size; i++) {
+            if (elements[i].getKey() == key
+                    || elements[i].getKey() != null && elements[i].getKey().equals(key)) {
+                return (V) elements[i].getValue();
+            }
+        }
         return null;
     }
 
     @Override
     public int size() {
-        return -1;
+        return size;
+    }
+
+    public Node[] getElements() {
+        return elements;
+    }
+
+    public void setElements(Node[] elements) {
+        this.elements = elements;
+    }
+
+    public int getSize() {
+        return size;
+    }
+
+    public void setSize(int size) {
+        this.size = size;
+    }
+
+    class Node<K, V> {
```

Comment:
```suggestion
    private class Node<K, V> {
```
this class should be private because only StorageImpl should be able to use it